src/Snake.java

```diff
@@ -0,0 +1,272 @@
+import java.util.ArrayList;
+import java.util.List;
+import edu.macalester.graphics.GraphicsGroup;
+import edu.macalester.graphics.GraphicsObject;
+import edu.macalester.graphics.Point;
+
+
+public class Snake extends GraphicsGroup {
+
+    private int i = 0;
+    private double speed = 2;
+
+    private double dx = 0;
+    private double dy = 0;
+
+    private static double width = 15;
+
+    private boolean movingLeft;
+    private boolean movingRight;
+    private boolean movingUp;
+    private boolean movingDown;
```

Comment:
Folding these four booleans into two dx/dy ints would simplify quite a lot of code below.